webui/react/src/components/Metadata.tsx

```diff
@@ -0,0 +1,63 @@
+import Button from 'hew/Button';
+import Icon from 'hew/Icon';
+import Message from 'hew/Message';
+import Tree, { TreeDataNode } from 'hew/Tree';
+import { useCallback, useMemo } from 'react';
+
+import { JsonObject, TrialDetails } from 'types';
+import { downloadText } from 'utils/browser';
+import { isJsonObject } from 'utils/data';
+
+import Section from './Section';
+
+export interface Props {
+  trial?: TrialDetails;
+}
+
+export const EMPTY_MESSAGE = 'No metadata found';
+
+const Metadata: React.FC<Props> = ({ trial }: Props) => {
+  const getNodes = useCallback((data: JsonObject): TreeDataNode[] => {
```

Comment:
given that this function isn't dependent on any props, we can drop the `useCallback` call by simply moving it out of the function.